### ISSUE TITLE
Get rid of or document extra menu items in "View" menu under macOS

### ISSUE THREAD

**Issue by vadz** [wxWidgets/wxWidgets#22210]
macOS seems to handle menu called "View" specially now (12.x): it adds some standard items to it as can be seen with
```diff
diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 470e765423..cbad4f73a5 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -159,6 +159,10 @@ MyFrame::MyFrame(const wxString& title)
     menuBar->Append(fileMenu, "&File");
     menuBar->Append(helpMenu, "&Help");

+    wxMenu *viewMenu = new wxMenu();
+    viewMenu->Append(wxID_ANY, "Toggle Sidebar");
+    menuBar->Append(viewMenu, "&View");
+
     // ... and attach this menu bar to the frame
     SetMenuBar(menuBar);
 #else // !wxUSE_MENUBAR
```
which results in something like this: 
<img width="478" alt="macos_view_menu" src="https://user-images.githubusercontent.com/146917/159168993-fb49a8e9-33fe-48b4-a140-1e37ed9a9730.png">

(see [this question](https://stackoverflow.com/questions/71509445/how-do-i-get-rid-of-the-default-macos-menu-items-in-wxwidgets)).

If we're fine with this -- as I guess this is the UI standard under Mac -- we need to at least document this behaviour, but ideally also provide some way to react to and maybe customize these menu items because I don't think neither tabs nor side bar make sense for _all_ applications. Also, toggling tab bar does show it currently, but I don't think it's very useful without some way of adding tabs to it. And toggling side bar does nothing at all.

Personally I think that "Full view" could be added by default (it should do no harm, at least), but the other two should be opt-in.

@csomor Any thoughts?

**Comment by csomor** [comment]
@vadz could you please retest, I've turned them off at the moment, we'd really have to find a proper way to wire them, making sure apps are really ready to handle this when exposing the functionality in the menus ...

**Comment by imciner2** [comment]
There was already a way to disable the automatic tabbing introduced in https://github.com/wxWidgets/wxWidgets/pull/1674, so why do we need to introduce a different place to turn it off?

**Comment by csomor** [comment]
@imciner2 sorry I was not aware of this, and this bug report was current, you are absolutely right, there is no reason to duplicate things.

But I guess, since this happened, then the default should be that it's off, unless you turn it on, otherwise at least im IMHO it does not perform the 'least surprising' behaviour, so I'll call your method from app startup.

**Comment by vadz** [comment]
FWIW I agree that it should be turned off by default, I didn't realize this had such visible consequences when reviewing #1674.

**Comment by vadz** [comment]
Just to confirm, this is fixed by 86e485c8d824e29fa2e053b3c4ad17149b30de18 and 184a1fa244795f877343f403a9d5f5050f2fff84, thanks!
